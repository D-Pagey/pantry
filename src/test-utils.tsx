import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import React, { FC, ReactNode } from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { FirebaseContext } from './components/ProviderFirebase';

type AllTheProvidersProps = {
    children: ReactNode;
    customContext?: any;
};

const AllTheProviders: FC<AllTheProvidersProps> = ({ children, customContext }) => (
    <FirebaseContext.Provider
        value={{
            categories: [],
            deleteFoodItem: () => () => null,
            expiringCount: 0,
            isAuthed: false,
            isCheckingAuth: false,
            fridge: [],
            signOut: () => null,
            updateFridge: () => null,
            user: {
                email: null,
                name: null
            },
            ...customContext
        }}
    >
        <MemoryRouter>{children}</MemoryRouter>
    </FirebaseContext.Provider>
);

const customRender = (ui: any, customContext?: any, options?: any) =>
    render(ui, { wrapper: () => <AllTheProviders customContext={customContext}>{ui}</AllTheProviders>, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
