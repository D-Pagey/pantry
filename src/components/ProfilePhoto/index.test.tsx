import React from 'react';

import { render } from '../../test-utils';
import { UserDan } from '../../fixtures';
import { ProfilePhoto } from '.';

const props = {
    owner: UserDan
};

describe('ProfilePhoto component', () => {
    it('should render initials if photo not provided', () => {
        const noPhotoUser = {
            name: 'Dan Page',
            email: 'dan@google.com'
        };

        const { container, getByText } = render(<ProfilePhoto {...props} owner={noPhotoUser} />);
        expect(container.firstChild).toMatchSnapshot();
        getByText('DP');
    });

    it('should render a photo if url provided', () => {
        const { getByTestId } = render(<ProfilePhoto {...props} />);
        getByTestId('photo');
    });
});
