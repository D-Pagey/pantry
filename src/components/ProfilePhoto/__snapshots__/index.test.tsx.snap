// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePhoto component should render initials if photo not provided 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FCBA3B;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 60px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 60px;
}

.c1 {
  font-size: 1.2rem;
  font-weight: bold;
}

<div
  class="c0"
>
  <span
    class="c1"
  >
    DP
  </span>
</div>
`;
